Swiftfin/App/AppDelegate.swift

```diff
@@ -38,17 +38,37 @@ class AppDelegate: NSObject, UIApplicationDelegate {
         AppDelegate.orientationLock
     }
 
-    static func changeOrientation(_ orientation: UIInterfaceOrientationMask) {
+    static func enterPlaybackOrientation() {
+        AppDelegate.changeOrientation(.landscape)
+    }
 
-//        guard UIDevice.isPhone else { return }
-//
-//        Self.orientationLock = orientation
-//
-//        if #available(iOS 16, *) {
-//            let windowScene = UIApplication.shared.connectedScenes.first as? UIWindowScene
-//            windowScene?.requestGeometryUpdate(.iOS(interfaceOrientations: orientation))
-//        } else {
-//            UIDevice.current.setValue(UIInterfaceOrientation.landscapeRight.rawValue, forKey: "orientation")
-//        }
+    static func leavePlaybackOrientation() {
+        if UIDevice.isIPad {
+            AppDelegate.changeOrientation(.allButUpsideDown)
+        } else {
+            // On iPhone, users likely want to return to portrait mode after leaving playback.
+            // However, we don't want to lock them into portrait mode, so switch back after a delay.
+            AppDelegate.changeOrientation(.portrait)
+
+            // 0.25 seconds is about the time to switch from landscape to portrait.
+            // Changing orientation again too early will cause the top time/battery bar to remain
+            // at the side instead of moving up top, like it should.
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
+                AppDelegate.changeOrientation(.allButUpsideDown)
+            }
+        }
+    }
+
+    private static func changeOrientation(_ orientation: UIInterfaceOrientationMask) {
```

Comment:
I know @LePips had been trying an implementation with `PreferenceUIHostingController` and I tried to get that working, but I couldn't. If it had been working and you guys want to enforce that design, I'll revert these changes and let you guys implement it again.

I think changing orientation imperatively as opposed to declaratively on the Views is fine for now, since there are so few entry and exit points for the video player and not enforcing orientation is causing major problems on iPad.

Additionally, if we are going with this approach, I can clean up and delete the other commented-out code from the other approach if you'd like.

Reply:
I'm fine with this for now if you want to keep it in, however it isn't the final behavior that we want. We want iOS to be in portrait only except for the video players and maybe some other views down the road? iPadOS should always be anything but upside down. This more closely mirrors the Apple TV app.

I actually do intend to make the video player overlay friendlier on portrait, meaning revamping the Advanced Settings stuff entirely and making things pretty on the overlay, but I never got to that.

I do want to take another shot at the preference orientation solution sometime. You're correct that there are only a few points where this stuff matters so it is fairly maintainable but I guess I just like how things are more declarative 🤷‍♂️ .

Again, I'm fine with this now but don't be surprised if it eventually gets reworked a bit.

Reply:
I'm totally fine with these changes getting overridden. I just want to make it so that the main branch of this repo is in good enough condition that someone like me could come by, build the app, and just start using it as an end user happily.

I slightly prefer the preference orientation approach here too and I would've implemented it here if I were more experienced in Swift (and if a clean solution was possible) so I'm looking forward to seeing your take on it.

I'll let you guys handle the rest of the orientation behavior. I mean for these changes to be a quick and dirty fix to clunky iOS portrait UI and app-breaking iPadOS rotation issues, not a long term solution — especially to the iPad rotation issues.